pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -0,0 +1,391 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dpi
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/configmap"
+	"github.com/tigera/operator/pkg/render/common/meta"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DeepPacketInspectionNamespace = "tigera-dpi"
+	DeepPacketInspectionName      = "tigera-dpi"
+	DefaultMemoryLimitDPI         = "1Gi"
+	DefaultMemoryRequestDPI       = "100Mi"
+	DefaultCPULimitDPI            = "1"
+	DefaultCPURequestDPI          = "100m"
+)
+
+func DPI(
+	ids *operatorv1.IntrusionDetection,
+	installation *operatorv1.InstallationSpec,
+	nodeTLSSecret *corev1.Secret,
+	typhaTLSSecret *corev1.Secret,
+	typhaCAConfigMap *corev1.ConfigMap,
+	pullSecrets []*corev1.Secret,
+	openshift bool,
+	hasNoLicense bool,
+) render.Component {
+	return &dpi{
+		ids:              ids,
+		installation:     installation,
+		nodeTLSSecret:    nodeTLSSecret,
+		typhaTLSSecret:   typhaTLSSecret,
+		typhaCAConfigMap: typhaCAConfigMap,
+		pullSecrets:      pullSecrets,
+		openshift:        openshift,
+		hasNoLicense:     hasNoLicense,
+	}
+}
+
+type dpi struct {
+	ids              *operatorv1.IntrusionDetection
+	installation     *operatorv1.InstallationSpec
+	nodeTLSSecret    *corev1.Secret
+	typhaTLSSecret   *corev1.Secret
+	typhaCAConfigMap *corev1.ConfigMap
+	pullSecrets      []*corev1.Secret
+	openshift        bool
+	hasNoLicense     bool
+	dpiImage         string
+}
+
+func (d *dpi) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := d.installation.Registry
```

Comment:
Nit, but is there any point in putting these in variables as opposed to just using them directly?